addons/web/static/tests/core/class_patcher_tests.js

```diff
@@ -0,0 +1,967 @@
+                    ]);
+
+                    (new B()).f();
+
+                    assert.verifySteps([
+                        'A.constructor',
+                        'B.constructor',
+                        'A.f',
+                        'B.f',
+                    ]);
+                });
+
+                QUnit.test('inheriting a patchable class that has patch', function (assert) {
+                    assert.expect(12);
+
+                    const A = makePatchable(
+                        class {
+                            constructor() {
+                                assert.step('A.constructor');
+                            }
+                            f() {
+                                assert.step('A.f');
+                            }
+                        }
+                    );
+
+                    A.patch('patch', T =>
+                        class extends T {
+                            constructor() {
+                                super();
+                                assert.step('patch.constructor');
+                            }
+                            f() {
+                                super.f();
+                                assert.step('patch.f');
+                            }
+                        }
+                    );
+
+                    class B extends A {
+                        constructor() {
+                            super();
+                            assert.step('B.constructor');
+                        }
+                        f() {
+                            super.f();
+                            assert.step('B.f');
+                        }
+                    }
+
+                    (new A()).f();
+
+                    assert.verifySteps([
+                        'A.constructor',
+                        'patch.constructor',
+                        'A.f',
+                        'patch.f',
+                    ]);
+
+                    (new B()).f();
+
+                    assert.verifySteps([
+                        'A.constructor',
+                        'patch.constructor',
+                        'B.constructor',
+                        'A.f',
+                        'patch.f',
+                        'B.f',
+                    ]);
+                });
+
+                QUnit.test('patch inherited patchable class', function (assert) {
+                    assert.expect(10);
+
+                    const A = makePatchable(
+                        class {
+                            constructor() {
+                                assert.step('A.constructor');
+                            }
+                            f() {
+                                assert.step('A.f');
+                            }
+                        }
+                    );
+
+                    /**
+                     * /!\ WARNING /!\
+                     *
+                     * If you want to patch class B, make it patchable
```

Comment:
this can be protected against by a check in the `patch` method

Reply:
yes, we talked about it, I forgot